frontend/cli/cmd_deploy.go

```diff
@@ -37,5 +41,14 @@ func (d *deployCmd) Run(ctx context.Context, projConfig projectconfig.Config) er
 	if err != nil {
 		return err
 	}
-	return engine.BuildAndDeploy(ctx, d.Replicas, !d.NoWait)
+	tm := optional.None[time.Duration]()
+	if d.Timeout.Milliseconds() > 0 {
+		tm = optional.Some(d.Timeout)
+	}
```

Comment:
You can replace this with:

```suggestion
	tm := optional.Zero(d.Timeout)
```